### PR TITLE
Fix decodedString / encodedString methods in PINDiskCache.m to suppor…

### PR DESCRIPTION
**Pull request by jlukanta**
…t SDKs older than 7.0.

### CODE REVIEW

**Comment by jlukanta** [comment]
iOS 6.0 devices would crash without this fix

File: PINCache/PINDiskCache.m [hunk 1]
```diff
@@ -141,16 +141,40 @@ - (NSString *)keyForEncodedFileURL:(NSURL *)url
...
 - (NSString *)encodedString:(NSString *)string
 {
-    if (![string length])
+    if (![string length]) {
         return @"";
-   return [string stringByAddingPercentEncodingWithAllowedCharacters:[[NSCharacterSet characterSetWithCharactersInString:@".:/"] invertedSet]];
+    }
+    
+    if ([string respondsToSelector:@selector(stringByAddingPercentEncodingWithAllowedCharacters:)]) {
+        return [string stringByAddingPercentEncodingWithAllowedCharacters:[NSCharacterSet characterSetWithCharactersInString:@".:/"]];
```
Comment (garrettmoon) [inline PINCache/PINDiskCache.m]: shouldn't this be the invertedSet?

**Comment by garrettmoon** [comment]
CFURLCreateStringByAddingPercentEscapes is deprecated and thus will cause warnings. Can you think of a way to only add the code path in if the target supports iOS 6?

**Comment by jlukanta** [comment]
Good catch! Fixed!

**Comment by jlukanta** [comment]
In general, I think it's better to suppress the deprecated warning instead of checking for the OS version because it requires no knowledge of when the functions were deprecated in all supported OS-es (iPhone and Mac). It should be less prone to errors, and results in cleaner code (less checks for different types of OS-es).

But I am fine either way, and I can change it if you think the other approach is better.

**Comment by garrettmoon** [comment]
Nope, I didn't notice the pragmas. I like your way better! Thanks!

**Pull request merged**